File: desktop_application/src/main/java/ru/fullrest/mfr/plugins_configuration_utility/model/entity/Details.java
package ru.fullrest.mfr.plugins_configuration_utility.model.entity;

import com.fasterxml.jackson.annotation.JsonIgnore;
import lombok.Data;
import lombok.EqualsAndHashCode;
import lombok.ToString;

import javax.persistence.*;

/**
 * Created on 02.11.2018
 *
 * @author Alexey Plekhanov
 */
@EqualsAndHashCode(callSuper = true)
@Data
@Entity
@Table(name = "DETAILS", indexes = {
        @Index(name = "DETAILS_UNIQUE_INDEX", columnList = "RELEASE_ID, STORAGE_PATH", unique = true)
})
public class Details extends BaseEntity {

    @ManyToOne(fetch = FetchType.EAGER)
    @JoinColumn(name = "RELEASE_ID", nullable = false)
    @JsonIgnore
    @EqualsAndHashCode.Exclude
    @ToString.Exclude
    private Release release;

    @Column(name = "STORAGE_PATH", nullable = false)
    private String storagePath;

    @Column(name = "GAME_PATH", nullable = false)
    private String gamePath;

    @Column(name = "MD5")
    @JsonIgnore
    @EqualsAndHashCode.Exclude
    @ToString.Exclude
    private byte[] md5;
}

File: desktop/application/src/main/kotlin/ru/fullrest/mfr/plugins_configuration_utility/model/entity/Details.kt
package ru.fullrest.mfr.plugins_configuration_utility.model.entity

import com.fasterxml.jackson.annotation.JsonIgnore
import javax.persistence.*

@Entity
@Table(name = "DETAILS")
class Details(

    @JsonIgnore
    @ManyToOne(fetch = FetchType.EAGER)
    @JoinColumn(name = "RELEASE_ID", nullable = false)
    var release: Release?,

    @Column(name = "STORAGE_PATH", nullable = false)
    var storagePath: String,

    @Column(name = "GAME_PATH", nullable = false)
    var gamePath: String
) : BaseEntity()